### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by BNDong** [BNDong/Cnblogs-Theme-SimpleMemory#238] [General]
### CDN
以前的版本比较依赖 GitHub 和 jsdelivr，新版本开始支持将打包后的资源文件托管到自己的云资源。

### 配置
以前配置都是边做主题边加的没有整体结构的规划，现将配置重新处理，配置方式不变，但配置名称及位置会有一定变化。
以前的配置不兼容，升级此版本的朋友需要重新处理配置，详细配置信息请参考文档。

### 版本映射
1. 移除了该功能，这个功能的存在是为了解决解决CDN缓存的问题，现在有了更便捷的方案解决这个问题，详细会在文档进行说明。

### 菜单
1. 提升了自定义菜单排序的优先级
2. 自定义菜单现在支持点击显示/隐藏了
3. 增加了菜单列表点击显示/隐藏的速度
4. 细节调整

### 右下角导航
1. 公众号按钮更改为二维码按钮，更换了对应 icon
2. 主页现在可以显示赞赏与二维码按钮了
3. 优化了顶/踩按钮的处理逻辑，修复重复点击动画的问题
4. 导航按钮现在比较多，增加了显示/隐藏的控制按钮
5. 文章页增加了目录控制按钮
6. 新增配置，点击跳至底部可以自定义滚动到某个dom元素的位置

### 文章
1. 移除了文章各级标题的样式渲染，取消了自动编号的功能
2. 以前的 banner 动效比较耗费性能，现替换新的 banner 动效
3. 增加了文章标题动效
4. 调整了文章信息展现的格式，增加了文章文字总数和预计阅读时间
5. 重构了文章目录的代码逻辑，现在目录支持多级了(1~5)
6. 移除了文章目录过长时，一级目录以下自动折叠的功能
7. 新的顶踩按钮样式
8. 新的引用文本样式
9. 新的底部信息按钮样式

### 代码框
1. 新的代码框样式，支持mac风格的代码框样式（可配置）
2. 精简了代码块的HTML基础结构
3. 移除了代码滚动条样式的渲染
4. 优化了对评论中代码块的高亮渲染
5. 移除了 prettyprint 代码高亮插件，此版本以后不再支持  prettyprint 类型的渲染
6. 移除了修改代码字体和大小的配置，如果需要修改样式请自行修改基础样式文件

### 文章后缀
1. 新的文章后缀信息框样式

### 广告
1. 综合考虑，移除了隐藏广告的功能

### 书单
1. 新的书单样式
2. 增加了阅读日期和阅读进度的配置及样式
3. 现在评分支持半星的评分了

### 钩子
1. 重新处理了钩子，改变了基础传参

### 友链
1. 新增了友链页面

<hr /><em>This discussion was created from the release <a href='https://github.com/BNDong/Cnblogs-Theme-SimpleMemory/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by TanTaiwoo07** [comment]
怎么弄？？
